src/snowflake/snowpark/_internal/analyzer/select_statement.py

```diff
@@ -0,0 +1,749 @@
+                # So we need to put all operands before intersect into a single operand.
+                existing_set_operands = (
+                    SetOperand(SetStatement(*self.from_.set_operands)),
+                )
+                sub_statement = SetStatement(
+                    *(SetOperand(x.to_subqueryable(), operator) for x in selectables)
+                )
+                set_operands = (SetOperand(sub_statement.to_subqueryable(), operator),)
+            else:
+                existing_set_operands = self.from_.set_operands
+                sub_statement = SetStatement(
+                    *(SetOperand(x.to_subqueryable(), operator) for x in selectables)
+                )
+                set_operands = (SetOperand(sub_statement.to_subqueryable(), operator),)
+            set_statement = SetStatement(
+                *existing_set_operands, *set_operands, analyzer=self.analyzer
+            )
+        else:
+            set_operands = tuple(
+                SetOperand(x.to_subqueryable(), operator) for x in selectables
+            )
+            set_statement = SetStatement(
+                SetOperand(self.to_subqueryable()),
+                *set_operands,
+                analyzer=self.analyzer,
+            )
+        new = SelectStatement(analyzer=self.analyzer, from_=set_statement)
+        new._column_states = set_statement.column_states
+        return new
+
+    def limit(self, n: int, *, offset: int = 0) -> "SelectStatement":
+        new = copy(self)
+        new.from_ = self.from_.to_subqueryable()
+        new.limit_ = min(self.limit_, n) if self.limit_ else n
+        new.offset = (self.offset + offset) if self.offset else offset
+        return new
+
+
+class SetOperand:
+    def __init__(self, selectable: Selectable, operator: Optional[str] = None) -> None:
+        super().__init__()
+        self.selectable = selectable
+        self.operator = operator
+
+
+class SetStatement(Selectable):
+    def __init__(self, *set_operands: SetOperand, analyzer=None) -> None:
+        super().__init__(analyzer=analyzer)
+        self.analyzer = analyzer
+        self.set_operands = set_operands
+        for operand in set_operands:
+            if operand.selectable.pre_actions:
+                if not self.pre_actions:
+                    self.pre_actions = []
+                self.pre_actions.extend(operand.selectable.pre_actions)
+            if operand.selectable.post_actions:
+                if not self.post_actions:
+                    self.post_actions = []
+                self.post_actions.extend(operand.selectable.post_actions)
+
+    def sql_query(self) -> str:
+        sql = f"({self.set_operands[0].selectable.sql_query()})"
+        for i in range(1, len(self.set_operands)):
+            sql += f" {self.set_operands[i].operator} ({self.set_operands[i].selectable.sql_query()})"
+        return sql
+
+    def schema_query(self) -> str:
+        return self.set_operands[0].selectable.schema_query()
```

Comment:
It takes me some time to understand why just using the first set operator's schema query. It's worth a comment here https://docs.snowflake.com/en/sql-reference/operators-query.html#general-usage-notes